src/Sdk/DTWebApi/WebApi/TaskAgentHttpClient.cs

```diff
@@ -104,7 +105,30 @@ public class TaskAgentHttpClient : TaskAgentHttpClientBase
                 LockedUntil = expiresOn,
             };
 
-            return UpdateAgentRequestAsync(poolId, requestId, lockToken, request, userState, cancellationToken);
+            var additionalHeaders = new Dictionary<string, string>();
+            if (!string.IsNullOrEmpty(orchestrationId))
+            {
+                additionalHeaders["X-VSS-OrchestrationId"] = orchestrationId;
```

Comment:
this Header will make everything on the service for this request has the same orchestrationid (activitylog/producttrace)